pkg/sessionctx/stmtctx/stmtctx.go

```diff
@@ -798,120 +800,137 @@ func (sc *StatementContext) AddAffectedRows(rows uint64) {
 		// For compatibility with MySQL, not add the affected row cause by the foreign key trigger.
 		return
 	}
-	sc.mu.Lock()
-	defer sc.mu.Unlock()
+	mu := &sc.mu
```

Comment:
I can't get the point that why using this way can avoid panic, any reference for mentioning it? From my persepctive, the panic may be related to other places.

Reply:
I wasn’t fully aware of the direct cause at that point. Your observation is indeed correct.
The direct cause was the Reset() function, which replaces sc itself, and consequently replaces mu.

As mentioned in this comment, how about locking the Mutex in Reset()?

https://github.com/pingcap/tidb/issues/58600#issuecomment-2566269116